File: f77/build05/def_data.f
C
        Subroutine Default_Data
C
C       This subroutine set a the default data values.
C
C       Inputs:
C           None.
C
C       Output:
C           Coef_Of_Restitution - Coefficent of restitution [Real]
C           DtMax               - Maximum allowable simulation time step [sec]
C           DtMin               - Minimum allowable simulation time step [sec]
C           DtPrint             - Time step between printing data [sec]
C           Gravity             - Acceleration due to gravity [m/sec**2]
C           Time0               - Initial time [sec]
C           Tstop               - Simulation stop time [sec]
C           X_IC                - Initial position of ball [m]
C           XD_IC               - Initial velocity of ball [m/sec]
C
C       Internal variables and constants:
C           None.
C
C       Declare global common
C
        Include '../driver/global.inc'
        Include '../driver/sysvars.inc'
        Equivalence(Real_Array(10), Coef_Of_Restitution)
        Equivalence(Real_Array(11), Gravity)
        Equivalence(Real_Array(12), X_IC)
        Equivalence(Real_Array(13), Xd_IC)
C
C       Set model independent data and reset time
C
        Dtmax   = 0.005e0
        DtMin   = 0.001e0
        DtPrint = 0.01e0
        Time0   = 0.0e0
        Tstop   = 10.0e0
C
C       Set the default data for the simulation
C
        Coef_Of_Restitution =  0.80e0
        Gravity             =  9.88e0
        X_IC                = 10.0e0
        Xd_IC               =  0.0e0
C
        Return
        End

File: gcc/build05/def_data.c
#include "def_data.h"
#include "../driver/global.h"
#include "../driver/sysvars.h"

#define coef_of_restitution real_array[10].r // Coefficent of restitution [Real]
#define gravity             real_array[11].r // Acceleration due to gravity [m/sec**2]
#define x_ic                real_array[12].r // Initial position of ball [m]
#define xd_ic               real_array[13].r // Initial velocity of ball [m/sec]

void default_data(void) {
  // Set model independent data and reset time
  dtmax = 0.005;
  dtmin = 0.001;
  dtprint = 0.01;
  time0 = 0.0;
  tstop = 10.0;
  // Set the default data for the simulation
  coef_of_restitution = 0.8;
  gravity = 9.88;
  x_ic = 10.0;
  xd_ic = 0.0;
}
